### PR TITLE
http_ext: Replace request to httpbin with mock

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

The request to httpbin service was failing with a 504 for me, so I replaced it with a mock.

The mock is unfortunately HTTP only, so if the intent was to test HTTPS as well, then this PR removes it.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/tedge_http_ext/Cargo.toml
Comment (didier-wenzek) [inline crates/extensions/tedge_http_ext/Cargo.toml:23]: This is a dev dependency.

**Review by didier-wenzek — COMMENTED** [review]
> The mock is unfortunately HTTP only, so if the intent was to test HTTPS as well, then this PR removes it.

Yes, the point was to test HTTPS connection. But it's okay to removed it as this test is indeed flaky due to its dependency to an external resource.

So okay to merge this PR, once fix the dependency.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Comment by reubenmiller** [comment]
Merging as this issue was blocking the release pipeline.

**Pull request merged**
